src/OpenLoco/src/World/Company.cpp

```diff
@@ -335,6 +335,139 @@ namespace OpenLoco
         company->observationTimeout = 5;
     }
 
+    /* 0x004A6841
+     * Creates a vector of all the available rail track (trains and trams) for a company
+     * Tram track is marked with a (1<<7) flag within the uint8_t
+     */
+    sfl::static_vector<uint8_t, Limits::kMaxRoadObjects + Limits::kMaxTrackObjects> companyGetAvailableRailTracks(const CompanyId id)
+    {
+        sfl::static_vector<uint8_t, Limits::kMaxRoadObjects + Limits::kMaxTrackObjects> result;
+        const auto* company = CompanyManager::get(id);
+        sfl::static_unordered_flat_set<uint8_t, Limits::kMaxTrackObjects> tracks;
+        for (auto i = 0u; i < ObjectManager::getMaxObjects(ObjectType::vehicle); ++i)
```

Comment:
I think you added `forEachLoadedObject` so you could make use  of that here.

Reply:
`forEachLoadedObject` is for all types of loaded object we only want to iterate a single type of loaded object. I've an issue for creating a helper function for this.